src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -1803,95 +1789,143 @@ HRESULT CorProfiler::GenerateVoidILStartupMethod(const ModuleID module_id,
+    //
+    ILRewriter rewriter_already_loaded(this->info_, nullptr, module_id, alreadyLoadedMethodToken);
+    rewriter_already_loaded.InitializeTiny();
+
+    ILInstr* pALFirstInstr = rewriter_already_loaded.GetILList()->m_pNext;
+    ILInstr* pALNewInstr = NULL;
+
+    // ldsflda _isAssemblyLoaded : Load the address of the "_isAssemblyLoaded" static var
+    pALNewInstr = rewriter_already_loaded.NewILInstr();
+    pALNewInstr->m_opcode = CEE_LDSFLDA;
+    pALNewInstr->m_Arg32 = isAssemblyLoadedFieldToken;
+    rewriter_already_loaded.InsertBefore(pALFirstInstr, pALNewInstr);
+
+    // ldc.i4.1 : Load the constant 1 (int) to the stack
+    pALNewInstr = rewriter_already_loaded.NewILInstr();
+    pALNewInstr->m_opcode = CEE_LDC_I4_1;
+    rewriter_already_loaded.InsertBefore(pALFirstInstr, pALNewInstr);
+
+    // ldc.i4.0 : Load the constant 0 (int) to the stack
+    pALNewInstr = rewriter_already_loaded.NewILInstr();
+    pALNewInstr->m_opcode = CEE_LDC_I4_0;
+    rewriter_already_loaded.InsertBefore(pALFirstInstr, pALNewInstr);
+
+    // call int Interlocked.CompareExchange(ref int, int, int) method
+    pALNewInstr = rewriter_already_loaded.NewILInstr();
+    pALNewInstr->m_opcode = CEE_CALL;
+    pALNewInstr->m_Arg32 = interlocked_compare_member_ref;
+    rewriter_already_loaded.InsertBefore(pALFirstInstr, pALNewInstr);
+
+    // ldc.i4.1 : Load the constant 1 (int) to the stack
+    pALNewInstr = rewriter_already_loaded.NewILInstr();
+    pALNewInstr->m_opcode = CEE_LDC_I4_1;
+    rewriter_already_loaded.InsertBefore(pALFirstInstr, pALNewInstr);
+
+    // ceq : Compare equality from two values from the stack
+    pALNewInstr = rewriter_already_loaded.NewILInstr();
+    pALNewInstr->m_opcode = CEE_CEQ;
+    rewriter_already_loaded.InsertBefore(pALFirstInstr, pALNewInstr);
+
+    // ret : Return the value of the comparison
+    pALNewInstr = rewriter_already_loaded.NewILInstr();
+    pALNewInstr->m_opcode = CEE_RET;
+    rewriter_already_loaded.InsertBefore(pALFirstInstr, pALNewInstr);
+
+    hr = rewriter_already_loaded.Export();
+    if (FAILED(hr)) {
+      Warn("GenerateVoidILStartupMethod: Call to ILRewriter.Export() failed for ModuleID=", module_id);
+      return hr;
+    }
+
+  } else {
+
+    /////////////////////////////////////////////
+    // Add IL instructions into the IsAlreadyLoaded method
+    //
+    //  public static bool IsAlreadyLoaded() {
+    //      return false;
+    //  }
+    //
+    ILRewriter rewriter_already_loaded(this->info_, nullptr, module_id, alreadyLoadedMethodToken);
+    rewriter_already_loaded.InitializeTiny();
+
+    ILInstr* pALFirstInstr = rewriter_already_loaded.GetILList()->m_pNext;
+    ILInstr* pALNewInstr = NULL;
+
+    // ldc.i4.0 : Load the constant 0 (int) to the stack
+    pALNewInstr = rewriter_already_loaded.NewILInstr();
+    pALNewInstr->m_opcode = CEE_LDC_I4_0;
+    rewriter_already_loaded.InsertBefore(pALFirstInstr, pALNewInstr);
+
+    // ret : Return the value of the comparison
```

Comment:
Is this comment correct? What comparison?

Reply:
Right, fixed in 4676405